crates/torii/libp2p/src/server/mod.rs

```diff
@@ -233,6 +252,60 @@ impl Relay {
     }
 }
 
+// Validates the message model 
+// and returns the identity and signature
+fn validate_message(message: &Ty) -> Result<(FieldElement, Vec<FieldElement>), Error> {
+    let mut identity: FieldElement = FieldElement::ZERO;
+    let mut signature: Vec<FieldElement> = Vec::new();
+    let message_struct = match message {
+        Ty::Struct(message) => {
+            for member in message.keys() {
+                match member.name.as_str() {
+                    "identity" => {
+                        // check if identity is correct primitive type
+                        if let Ty::Primitive(Primitive::ContractAddress(identity)) = &member.value {
+                            identity = &identity.clone();
+                        } else {
+                            return Err(Error::InvalidMessageError("Identity is not a contract address".to_string()));
+                        }
+                    }
+                    "signature" => {
+                        // must be a Ty::Tuple and children must be Primitive::Felt252
+                        if let Ty::Tuple(signature) = &member.value {
+                            for component in signature {
+                                if let Ty::Primitive(Primitive::Felt252(sig)) = &member.value {
+                                    signature.push(sig.clone());
+                                } else {
+                                    return Err(Error::InvalidMessageError("Signature component is not a Felt252".to_string()));
+                                }
+                            }
+                        } else {
+                            return Err(Error::InvalidMessageError("Signature is not a tuple".to_string()));
+                        }
+                    }
+                    _ => {}
+                }
+            }
+
+            signature = Signature::new(sig_r, sig_s);
+            message
+        }
+        _ => {
+            return Err(Error::InvalidMessageError("Message is not a struct".to_string()));
+        }
+    };
+
+    if identity == FieldElement::ZERO {
+        return Err(Error::InvalidMessageError("Missing identity".to_string()));
+    }
+
+    if signature.is_empty() {
```

Comment:
If we only support signature with 2 felts, we may ensure the length is exactly two.